packages/dev-frontend/src/components/Bonds/context/api.ts

```diff
@@ -79,9 +87,11 @@ const getAccountBonds = async (
       const breakEvenTime = getFutureTimeByDays(toFloat(breakEvenDays) - bondAgeInDays);
       const rebondTime = getFutureTimeByDays(toFloat(rebondDays) - bondAgeInDays);
       const marketValue = decimalify(bondAccrueds[idx]).mul(marketPrice);
-      const claimNowReturn = getReturn(accrued, deposit, marketPrice);
-      const rebondReturn = getReturn(rebondAccrual, deposit, marketPrice);
-      const rebondRoi = Decimal.from(rebondReturn).div(deposit);
+
+      // Accrued bLUSD is 0 for cancelled/claimed bonds
+      const claimNowReturn = accrued.isZero ? "0" : getReturn(accrued, deposit, marketPrice);
+      const rebondReturn = accrued.isZero ? "0" : getReturn(rebondAccrual, deposit, marketPrice);
+      const rebondRoi = accrued.isZero ? Decimal.ZERO : Decimal.from(rebondReturn).div(deposit);
```

Comment:
`rebondReturn` can be negative when there's no premium, in which case `Decimal.from(...)` will throw because it doesn't support negatives.